packages/core/src/verification/signature-verifier.ts

```diff
@@ -0,0 +1,543 @@
+            return {
+              gateway,
+              dataItemOffset: parseInt(dataItemOffset, 10),
+              dataItemDataOffset: parseInt(dataItemDataOffset, 10),
+              rootTransactionId,
+              dataItemSize: parseInt(dataItemSize, 10),
+            };
+          }
+
+          // if we get here, we didn't get the information we need from this gateway
+          throw new Error('Missing data item attributes from trusted gateway', {
+            cause: {
+              txId,
+              gateway: gateway.toString(),
+              dataItemOffset,
+              dataItemDataOffset,
+              rootTransactionId,
+              dataItemSize,
+            },
+          });
+        });
+      },
+    );
+
+    const {
+      gateway,
+      dataItemOffset,
+      dataItemDataOffset,
+      rootTransactionId,
+      dataItemSize,
+    } = await Promise.any(dataItemAttributesPromises);
+    this.logger.debug('Successfully fetched data item attributes for txId', {
+      txId,
+      dataItemOffset,
+      dataItemDataOffset,
+      rootTransactionId,
+      dataItemSize,
+      gateway: gateway.toString(),
+    });
+    return {
+      dataItemOffset,
+      dataItemDataOffset,
+      rootTransactionId,
+      dataItemSize,
+    };
+  }
+
+  /**
+   * Fetches just the bytes needed for signature verification from the root transaction
+   *
+   * @param txId - The transaction ID to get signature data for
+   * @returns The bytes containing the data item up to the data section
+   */
+  async getSignatureData({
+    txId,
+  }: {
+    txId: string;
+  }): Promise<{
+    signatureType: number;
+    signature: Uint8Array;
+    owner: Uint8Array;
+    target: Uint8Array;
+    anchor: Uint8Array;
+    tags: Uint8Array;
+  }> {
+    const { rootTransactionId, dataItemOffset, dataItemDataOffset } =
+      await this.getDataItemAttributes(txId);
+
+    // Calculate the byte range for the request
+    const rangeStart = dataItemOffset;
+    const rangeEnd = dataItemDataOffset - 1;
+
+    // TODO: we are fetching data here via range request - we may not want to do this concurrently and instead shuffle gateways before iterating through each
+    for (const gateway of this.trustedGateways) {
+      try {
+        const url = `${gateway.toString()}${rootTransactionId}`;
+        const response = await fetch(url, {
+          method: 'GET',
+          redirect: 'follow',
+          mode: 'cors',
+          headers: {
+            Range: `bytes=${rangeStart}-${rangeEnd}`,
+            'Cache-Control': 'no-cache',
+          },
+        });
+
+        if (!response.ok || response.body === null) {
+          continue;
+        }
+
+        // create the data item object from just the signature bytes, so we can get the components easily
+        const trustedDataItemHeaderBytes = Buffer.from(
+          await response.arrayBuffer(),
+        );
+        const dataItem = new DataItem(trustedDataItemHeaderBytes);
```

Comment:
Very interesting that this works and does not enforce internal consistency of the data item on the passed in buffer

Reply:
added a comment that this is a hack for easily parsing out signature, owner, etc. from the header data